system/src/main.cpp

```diff
@@ -154,72 +148,73 @@ void system_handle_single_click()
-        system_prepare_display_bars();
-        int rssi = 0;
-        int bars = 0;
+    system_prepare_display_bars();
+    int rssi = 0;
+    int bars = 0;
 #if Wiring_WiFi == 1
-        rssi = WiFi.RSSI();
+    rssi = WiFi.RSSI();
 #elif Wiring_Cellular == 1
-        CellularSignal sig = Cellular.RSSI();
-        rssi = sig.rssi;
+    CellularSignal sig = Cellular.RSSI();
+    rssi = sig.rssi;
 #endif
-        if (rssi < 0) {
-            if (rssi >= -57) bars = 5;
-            else if (rssi > -68) bars = 4;
-            else if (rssi > -80) bars = 3;
-            else if (rssi > -92) bars = 2;
-            else if (rssi > -104) bars = 1;
-        }
-        DEBUG("RSSI: %ddB BARS: %d\r\n", rssi, bars);
-
-        system_display_bars(bars);
+    if (rssi < 0) {
+        if (rssi >= -57) bars = 5;
+        else if (rssi > -68) bars = 4;
+        else if (rssi > -80) bars = 3;
+        else if (rssi > -92) bars = 2;
+        else if (rssi > -104) bars = 1;
     }
+    DEBUG("RSSI: %ddB BARS: %d\r\n", rssi, bars);
+
+    system_display_bars(bars);
 }
 
-void system_handle_double_click()
+void system_handle_button_click()
 {
-    SYSTEM_POWEROFF = 1;
-    network.connect_cancel(true, true);
+    const uint8_t clicks = button_final_clicks;
+    button_final_clicks = 0;
+    switch (clicks) {
+    case 1: // Single click
+        system_display_rssi();
+        break;
+    case 2: // Double click
+        SYSTEM_POWEROFF = 1;
+        network.connect_cancel(true, true);
+        break;
+    default:
+        break;
+    }
 }
+#endif // #if Wiring_SetupButtonUX
 
-void reset_button_click(uint32_t release_time)
+void reset_button_click()
 {
-    clicks = 0;
-    prev_release_time = release_time - 1000;
+    const uint8_t clicks = button_current_clicks;
+    button_current_clicks = 0;
     CLR_BUTTON_TIMEOUT();
+    if (clicks > 0) {
+        system_notify_event(button_final_click, clicks);
+        button_final_clicks = clicks;
+    }
 }
 
-void handle_button_click(uint16_t depressed_duration, uint32_t release_time)
+void handle_button_click(uint16_t depressed_duration)
 {
-    uint32_t start_time = release_time - depressed_duration;
-    uint32_t since_prev = start_time - prev_release_time;
     bool reset = true;
-    if (depressed_duration<500) {               // a short button press
-        if (!clicks || (since_prev<1000)) {		// first click or within 1 second of the previous click
-            clicks++;
-            prev_release_time = release_time;
-            if (clicks == 1)
-            {
-                // If a second button press doesn't come within 1.5 seconds,
-                // declare a single button press valid.
-                ARM_BUTTON_TIMEOUT(1500);
-                reset = false;
-            }
-            else if (clicks == 2)
-            {
-                reset_button_click(release_time);
-                system_handle_double_click();
-            }
+    if (depressed_duration < 500) { // a short button press
```

Comment:
Looks great @sergeuz thank you for simplifying this.  I tested the PR and it works well.  It is rare but possible to get a SETUP button press to "bounce" and cause a soft power down or potentially 3-5 clicks, when you only wanted RSSI. (for that matter, getter a larger number of clicks than intended). To mitigate this we could add a lower bounds to the depressed_duration.  I played with 30ms which is almost impossible to manually push the button faster than, but at the same time provides a fairly decent debounce time.
`if (depressed_duration > 30 && depressed_duration < 500) { // a short button press`
If the press is outside this range (30-500), it is ignored. Thoughts?